components/common/src/command/package/download.rs

```diff
@@ -0,0 +1,384 @@
+//! Downloads a Habitat package from a [depot](../depot).
+//!
+//! # Examples
+//!
+//! ```bash
+//! $ hab pkg download core/redis
+//! ```
+//!
+//! Will download `core/redis` package from a custom depot:
+//!
+//! ```bash
+//! $ hab pkg install core/redis/3.0.1 redis -u http://depot.co:9633
+//! ```
+//!
+//! This would install the `3.0.1` version of redis.
+//!
+//! # Internals
+//!
+//! * Download the artifact
+//! * Verify it is un-altered
```

Comment:
Verification is done on install, we should not need to do it on download (it is an expensive operation)